app/Services/Pocket.php

```diff
@@ -0,0 +1,221 @@
+    {
+
+        try {
+            $response = $this->getHttpClient()->post("get", array(
+                'headers' => [
+                    'Content-Type' => 'application/json',
+                    'X-Accept' => 'application/json'
+                ],
+                "body" => json_encode(
+                    array(
+                        "consumer_key" => env('POCKET_API'),
+                        "access_token" => $this->user->access_token,
+                        "contentType"  => "article",
+                        "status"       => "unread",
+                        "sort"         => "newest",
+                        "detailType"   => "simple"
+                    )
+                )
+            ));
+
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+            return array();
+        }
+
+        $statusCode = $response->getStatusCode();
+        if ($statusCode != 200) {
+            if ($statusCode == 400) {
+                Log::error('Invalid pocket request');
+            } elseif($statusCode == 401) {
+                Log::error('Problem authenticating user');
+            } elseif($statusCode == 403) {
+                Log::error('Authenticated but lack of permission to access this resource');
+            } elseif($statusCode == 500) {
+                Log::error('Pocket server issue');
+            }
+        }
+
+        $feed = $response->json();
+
+        return $feed['list'];
+    }
+
+    /**
+     * Add user item row for an item if it doesn't already exist
+     *
+     * @param $item
+     */
+    protected function addUserItem($item) {
+        $userItem = UserItemsModel::where('user_id', '=', $this->user->id)
+            ->where('item_id', '=', $item->id)
+            ->limit(1)
+            ->get()
+            ->first();
+
+        if ($userItem == null) {
+            $userItem = new UserItemsModel();
+            $userItem->user_id = $this->user->id;
+            $userItem->item_id = $item->id;
+            $userItem->save();
+        }
+    }
+
+    protected function addItem($url, $title = null, $excerpt = null)
+    {
+        if (ItemModel::find(md5($url))) {
+            return ItemModel::find(md5($url));
+        }
+
+        if (!$articleText = $this->getArticleContent($url)) {
+            Log::error('Unable to find article for URL: ' . $url);
+        }
+
+        $item = new ItemModel();
+        $item->id = md5($url);
+        $item->url = $url;
+        $item->content = $this->replace4byte($articleText);
+        $item->title = $this->replace4byte($title);
+        $item->excerpt = $this->replace4byte(substr($excerpt, 0, 255));
+        $item->status = ItemModel::STATUS_FETCHED;
+        $item->save();
+
+        return $item;
+    }
+
+    /**
+     * Get article body text
+     * First try Goose, then try Readability
+     *
+     * @param $url
+     * @return string
+     */
+    protected function getArticleContent($url)
+    {
```

Comment:
I don't really agree that synchronising Pocket and fetching article text should be the same task. If we get items from Pocket, but fail to fetch their text (because a service is down or something) then we'll have to get items from Pocket again.